### DISCUSSION TITLE
[SOLVED] Add URLs programatically without a client

### DISCUSSION THREAD

**Discussion by shtrom** [wallabag/wallabag#6228] [General]
Ahoy,

tl;dr: How can I add new URLs to Wallabag programatically, i.e., without a client. I'm hopeful for a Symfony console approach, or some pithy cURL requests, that I can run in a shell (or similar) loop?

I have a bunch of URLs (which may or may not come from a dump of my Android DB after a disk dying on my server), that I want to add to a Wallabag instance (which may or may not have been restored from a previous backup).

I have played around in the Android app trying to add entries to the `QUEUE_ITEM` table, but didn't succeed in making those getting re-imported. I came to the conclusion that all I needed was simply to re-add the URLs, and let Wallabag do its thing.

So, how can I add a bunch of URLs from a simple list into Wallabag without using a UI client?

EDIT: Perhaps the `wallabag:import` command could work, with a manually recreated JSON?

**Comment by nicosomb** [comment]
Hello @shtrom. 

Yes, you can use `wallabag:import` command. You can have a look to `tests/Wallabag/ImportBundle/fixtures/wallabag-v2-read.json` to have an example.

**Comment by shtrom** [comment]
Ooook. So despite some warnings, this seems to have worked:

Generate the list of URLs from the Android DB
```
sqlite3 wallabag -json " select _id as id, url as url, strftime('%Y-%m-%dT%H:%M:%S+1000', creation_date/1000, 'unixepoch') as created_at from article where ifnull(article_id, '') is ''" > wallabag_v2.json
```

(I'm using the docker image, so) copy the file into the container
```
docker cp wallabag_v2.json wallabag_wallabag_1:/tmp
```

Then run the import
```
docker-compose exec wallabag /var/www/wallabag/bin/console wallabag:import shtrom /tmp/wallabag_v2.json --env=prod --importer=v2
```
[Note that this may cause a permissions issue on the `images` directory as the images from the newly imported articles would be owned by `root`, and cause a `PHP message: PHP Warning:  mkdir(): Permission denied in /var/www/wallabag/src/Wallabag/CoreBundle/Helper/DownloadImages.php on line 267" while reading response header from upstream`; this is fixed with a `chown -R nobody:nogroup` on the `images` directory]

That worked 👌🏼 Thanks for the pointer @nicosomb 

(The warnings were from missing title, content, mimetype and other, but that didn't seem to cause an issue. A few `isset` would help silencing those).

**Comment by nicosomb** [comment]
Thank you for the details. If you want / can, don’t hesitate to improve the code 😊
